File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/profiles/SharedProfileRepository.kt
package com.reguerta.user.domain.profiles

interface SharedProfileRepository {
    suspend fun getAllSharedProfiles(): List<SharedProfile>

    suspend fun getSharedProfile(userId: String): SharedProfile?

    suspend fun upsertSharedProfile(profile: SharedProfile): SharedProfile

    suspend fun deleteSharedProfile(userId: String): Boolean
}

File: ios/Reguerta/Reguerta/Domain/Profiles/SharedProfileRepository.swift
import Foundation

protocol SharedProfileRepository: Sendable {
    func allSharedProfiles(environment: SessionEnvironment) async throws -> [SharedProfile]
    func sharedProfile(userId: String, environment: SessionEnvironment) async throws -> SharedProfile?
    func upsert(profile: SharedProfile, environment: SessionEnvironment) async throws -> SharedProfile
    func deleteSharedProfile(userId: String, environment: SessionEnvironment) async throws -> Bool
}
